mnefun/_sss.py

```diff
@@ -577,16 +577,27 @@ def _pos_valid(pos, dist_limit, gof_limit):
 def _get_t_window(p, raw):
     t_window = p.coil_t_window if p is not None else 'auto'
     if t_window == 'auto':
-        from mne.chpi import _get_hpi_info
         try:
             hpi_freqs, _, _ = _get_hpi_info(raw.info, verbose=False)
+            info = raw.info
+            if info['line_freq'] is not None:  # as in mne.chpi.py
+                line_freqs = np.arange(info['line_freq'], info['sfreq'] / 3.,
+                                       info['line_freq'])
+            else:
+                line_freqs = np.array([60])   # SMB: should be safe to assume
         except RuntimeError:
             t_window = 0.2
         else:
-            # Use the longer of 5 cycles and the difference in HPI freqs.
-            # This will be 143 ms for 7 Hz spacing (old) and
-            # 60 ms for 83 Hz lowest freq.
-            t_window = max(5. / min(hpi_freqs), 1. / np.diff(hpi_freqs).min())
+            # Use the longer of 5 cycles and the difference across all
+            # HPI and line harmonic frequencies.
+            # This will be 143 ms for 7 Hz spacing (e.g. w/ higher coil freqs
+            # of 727 Hz+) and 83 ms for 60 Hz line freq (w/ more typical coil
+            # freqs of 83 Hz+).
+            all_freqs = np.concatenate((hpi_freqs, line_freqs))
+            delta_freqs = [abs(b - a) for b in all_freqs for a in all_freqs
+                           if a != b]
+            delta_freqs = np.array(delta_freqs)
```

Comment:
```suggestion
            delta_freqs = np.diff(np.unique(np.concatenate((hpi_freqs, line_freqs))))
```

Reply:
That's a nice simplification, because np.unique() puts the frequencies in order (so larger differences don't even have to be computed).  Thanks!